OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameMenuLogic.cs

```diff
@@ -88,14 +88,42 @@ public IngameMenuLogic(Widget widget, World world, Action onExit, WorldRenderer
 
 			abortMissionButton.OnClick = () =>
 			{
-				if (world.IsGameOver)
+				hideMenu = true;
+
+				if ((world.LocalPlayer == null || world.LocalPlayer.WinState != WinState.Won) &&
+								   (!world.IsGameOver || world.Map.Visibility == MapVisibility.MissionSelector))
```

Comment:
`if (world.LocalPlayer == null || (world.LocalPlayer.WinState != WinState.Won && (!world.IsGameOver || world.Map.Visibility == MapVisibility.MissionSelector)))`

Also a few tabs too much on the second line, it seems.